### ISSUE TITLE
`TypedDict` should extend `dict` not `Mapping`

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#260]
And it should override the `clear` method to be `Never`.

**Comment by DetachHead** [comment]
the only way to do this without breaking the functionality that bans `clear` from being called on `TypedDict`s would be to override its `clear` method with `clear: Never`, but that wouldn't work because pyright doesn't complain when you use objects typed as `Never`

so this change relies on #97

**Comment by DetachHead** [comment]
this would also make  `reportInvalidCast` less annoying ([see here](https://github.com/DetachHead/basedpyright?tab=readme-ov-file#note-about-casting-with-typeddicts))

**Comment by DetachHead** [comment]
eh, i don't think we should do this, it becomes less safe even if we make `clear` typed as `Never`:

```py
foo: dict[{"a": int}]

bar: dict[str, int] = foo

bar.clear() # no error
```

**Comment by KotlinIsland** [comment]
Is this only because we don't get an error about unreachable?

**Comment by DetachHead** [comment]
no it's because the `Never` is widened to something that's not `Never`
